### ISSUE TITLE
Zend 2 app + SendGrid emails sent are blocked by Gmail

### ISSUE THREAD

**Issue by mkykode** [JouwWeb/SlmMail#98]
I'm integrating SendGrid with Zend app and it works for all regular addresses except for gmail. :-1: 
SendGrid says it's not hitting their servers so not sure what is going on. Any help will be greatly appreciated.
The Zend 2 code I'm using is (obviously not posting the user and pass)

---

$smtpServer = 'smtp.sendgrid.net';  
$message = new Mail\Message();  
 $message->addTo($to)  
 ->addFrom('info@danceraccess.com')  
 ->setSubject($subject)  
 ->setBody($body);  
 // Setup SMTP transport using LOGIN authentication  
 $transport = new SmtpTransport();  
 $options = new SmtpOptions(array(  
 'name' => $smtpServer,  
 'host' => $smtpServer, //'127.0.0.1',  
 'port' => 25, // Notice port change for TLS is 587  
 'connection_class' => 'plain',  
 'connection_config' => array(  
 'username' => $username,  
 'password' => $password,  
 //'ssl' => 'tls',  
 ),  
 ));  
 $transport->setOptions($options);  
 $transport->send($message);

---

Here is the failed header when trying to send test message to SendGrid.

This message was created automatically by mail delivery software.
A message that you sent could not be delivered to one or more of its  
recipients. This is a permanent error. The following address(es) failed:

 sendgrid.tests@gmail.com  
 host alt4.gmail-smtp-in.l.google.com [74.125.24.27]  
 SMTP error from remote mail server after end of data:  
 550-5.7.1 [70.32.72.15 5] Our system has detected that this message is  
 550-5.7.1 likely unsolicited mail. To reduce the amount of spam sent to Gmail,  
 550-5.7.1 this message has been blocked. Please visit  
 550 5.7.1 https://support.google.com/mail/answer/188131 for more information. 4si2564792wmb.92 - gsmtp  
Reporting-MTA: dns; zsfs-bp7g.accessdomain.com

Action: failed  
Final-Recipient: rfc822;sendgrid.tests@gmail.com  
Status: 5.0.0  
Remote-MTA: dns; alt4.gmail-smtp-in.l.google.com  
Diagnostic-Code: smtp; 550-5.7.1 [70.32.72.15 5] Our system has detected that this message is  
550-5.7.1 likely unsolicited mail. To reduce the amount of spam sent to Gmail,  
550-5.7.1 this message has been blocked. Please visit  
550 5.7.1 https://support.google.com/mail/answer/188131 for more information. 4si2564792wmb.92 - gsmtp
From: info@danceraccess.com  
Subject: Your new password!  
Date: February 24, 2016 at 11:36:52 PM PST  
To: sendgrid.tests@gmail.com

---

Here is the failure report:
Event:  failure error  
Sender User:    danceraccess  
Sender Domain:  danceraccess.com  
Sender: info@danceraccess.com  
Sent Time:  Feb 24, 2016 11:36:23 PM  
Sender Host:    smtp.sendgrid.net  
Sender IP:  127.0.0.1  
Authentication: dovecot_plain  
Spam Score: 0  
Recipient:  sendgrid.tests@gmail.com  
Delivered To:  
Delivery User:  
Delivery Domain:  
Router: dkim_lookuphost  
Transport:  dkim_remote_smtp  
Out Time:   Feb 24, 2016 11:37:23 PM  
ID: 1aYqTk-00086I-EK  
Delivery Host:  alt4.gmail-smtp-in.l.google.com  
Delivery IP:    74.125.24.27  
Size:   5.94 KB  
Result: ECDHE-RSA-AES128-GCM-SHA256:128 CV=yes: SMTP error from remote mail server after end of data: 550-5.7.1 [70.32.72.15 5] Our system has detected that this message is\n550-5.7.1 likely unsolicited mail. To reduce the amount of spam sent to Gmail

---

Please help!!!

**Comment by bakura10** [comment]
Hi,

SlmMail was not made for sending using Smtp. SlmMail is a wrapper around official APIs.

You should use the SendGrid service to send your emails: https://github.com/juriansluiman/SlmMail/blob/master/src/SlmMail/Service/SendGridService.php

**Comment by mkykode** [comment]
Ok let me try to integrate this.

**Comment by mkykode** [comment]
Hi @bakura10 
I tried implementing your code but I keep getting the error:

Zend\ServiceManager\ServiceManager::get was unable to fetch or create an instance for SlmMail\Mail\Transport\SendGridTransport

On your page, I don't see $locator defined, so I changed it to this
 `     
        use SlmMail\Service;
        use SlmMail\Mail\Transport; 

        $mail = new \Zend\Mail\Message();
        $mail->setBody($body);
        $mail->setFrom('admin@'.$mailsrvr, 'DancerAccess');
        $mail->addTo($to);
        $mail->setSubject($subject);
        $locator = $this->getServiceLocator();
        $transport = $locator->get('SlmMail\Mail\Transport\SendGridTransport');
        $transport->send($mail);
`
Can you correct this?

**Comment by roelvanduijnhoven** [comment]
I am closing all old issues with no recent activity. Feel free to open a new issue, and we will look into the matter. Thanks!
